python/tvm/tir/ir_builder_v2.py

```diff
@@ -733,6 +744,23 @@ def wrapped(*args, **kwargs):
 # pylint: enable=invalid-name
 
 
+class inline:
+    def __init__(self, value) -> None:
+        self.value = value
+        self.i = 0
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if isinstance(self.value, (tuple, list)):
+            if self.i < len(self.value):
+                v = self.value[self.i]
+                self.i += 1
+                return inline(v)
+        raise StopIteration
```

Comment:
Instead let's use yield:

```suggestion
    def __iter__(self):
        def f():
            for i in self.value:
                yield i

        return f()

```